### ISSUE TITLE
[Bug]: Badge not working in iOS with "Increase by" option

### ISSUE THREAD

**Issue by devsideal** [OneSignal/OneSignal-Flutter-SDK#893]
### What happened?

Badge not working in iOS with "Increase by" option, alwasy shwoing only 1 count.

https://github.com/OneSignal/OneSignal-Flutter-SDK/assets/91250232/432e9410-bee2-447d-a193-aff73214150b

### Steps to reproduce?

```Markdown
1. Added package successfully
2. Followed all the steps given here https://documentation.onesignal.com/docs/badges
3. Send push from OneSignal console by setting 1 in "Increase by" option
4. In iPhone alwasy showing count 1 (for reference please see the video)
```

### What did you expect to happen?

If we send push with Increase by 1, It should add in previous count.

### OneSignal Flutter SDK version

onesignal_flutter: ^3.5.1

### Which platform(s) are affected?

- [X] iOS
- [ ] Android

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
@devsideal thanks for reaching out and providing reproduction steps! We're looking into this and will get back to you as soon as possible.

**Comment by jkasten2** [comment]
@devsideal This feature requires the Notification Service Extension. Can you double check you have it setup correctly based on the documentation? 
https://documentation.onesignal.com/docs/flutter-sdk-setup#add-notification-service-extension

**Comment by devsideal** [comment]
@jkasten2  Yes I have created the OneSignalNotificationServiceExtension and Added  App Group capability in both "Main App Target" and "OneSignalNotificationServiceExtension Target" as mintioned on the documentatoin.

**Comment by yuyangpl** [comment]
Make sure the "Set to" option is OK。
onesignal_flutter: 5.2.1
Flutter 3.22.2

**Comment by devsideal** [comment]
@yuyangpl Yes, "Set to" option is OK But "Increase by" is not working as expected.
and currenlly using `onesignal_flutter: ^3.5.1` and `Flutter 3.19.5`, I will try with updated versions.

**Comment by ngee147** [comment]
having the same issue

onesignal_flutter: ^5.2.1
Flutter 3.22.2

**Comment by devsideal** [comment]
Still not working with, i tried with updated version too, `onesignal_flutter: ^3.5.1` and `Flutter 3.19.5`

**Comment by yuyangpl** [comment]
> @yuyangpl Yes, "Set to" option is OK But "Increase by" is not working as expected. and currenlly using `onesignal_flutter: ^3.5.1` and `Flutter 3.19.5`, I will try with updated versions.

Hello, is there any progress on this question

**Comment by yuyangpl** [comment]
> @devsideal thanks for reaching out and providing reproduction steps! We're looking into this and will get back to you as soon as possible.

Hello, is there any progress on this question

**Comment by yuyangpl** [comment]
Hello, is this problem no longer tracked? The latest version still has this problem

**Comment by yuyangpl** [comment]
这个问题没人跟踪吗

**Comment by yuyangpl** [comment]
> 仍然不起作用，我也尝试了更新的版本，`onesignal_flutter: ^5.2.1`和`Flutter 3.22.2`

你好，请问这个问题你们已经解决了吗？

**Comment by nan-li** [comment]
Hi all, apologies for the delay. I am not able to reproduce this issue on my end.

1. Firstly, confirm you have set up the `OneSignalNotificationServiceExtension` **correctly**. You can check this by adding an image to your notification and seeing it does display, and by modifying this code below, adding the string `[Modified!!!!] ` to your notification. Then, check that your notification displays with the string `[Modified!!!!] ` before your text.

```swift
// Swift

if let bestAttemptContent = bestAttemptContent {
    // DEBUGGING: Uncomment the 2 lines below to check this extension is executing
    bestAttemptContent.body = "[Modified!!!!] " + bestAttemptContent.body

    OneSignalExtension.didReceiveNotificationExtensionRequest(self.receivedRequest, with: bestAttemptContent, withContentHandler: self.contentHandler)
}
```
```objc
// Objective-C
 self.bestAttemptContent = [request.content mutableCopy];

  // DEBUGGING: Uncomment the 2 lines below and comment out the one above to ensure this extension is executing            
  self.bestAttemptContent.body = [@"[Modified!!!!] " stringByAppendingString:self.bestAttemptContent.body];

  [OneSignalExtension didReceiveNotificationExtensionRequest:self.receivedRequest
                     withMutableNotificationContent:self.bestAttemptContent
                                 withContentHandler:self.contentHandler];
```

2. If the `OneSignalNotificationServiceExtension` is correctly setup, please confirm if you are sending it via the OneSignal dashboard or via the API.

3. If you are sending this notification via the API, please supply the entire payload (you can redact any sensitive information).

**Comment by yuyangpl** [comment]
OneSignal dashboard

![Image](https://github.com/user-attachments/assets/2dc4fc80-e018-4751-97bc-c12621def0ef)

**Comment by nan-li** [comment]
Hi @yuyangpl did you check that your `OneSignalNotificationServiceExtension` is working correctly? By changing the text in the `OneSignalNotificationServiceExtension` and see it is changed in the notification?

**Comment by yuyangpl** [comment]
*if* *let* bestAttemptContent = bestAttemptContent {

            /* DEBUGGING: Uncomment the 2 lines below to check this
extension is executing

             Note, this extension only runs when mutable-content is set

             Setting an attachment or action buttons automatically adds
this */

             print("Running NotificationServiceExtension")

             bestAttemptContent.body = "[Modified] " + bestAttemptContent.
body



            OneSignalExtension.didReceiveNotificationExtensionRequest(*self*
.receivedRequest, with: bestAttemptContent, withContentHandler: *self*.
contentHandler)

        }

body中并没有[modified],可能得原因是什么呢？

Nan ***@***.***> 于2025年6月10日周二 01:32写道：

**Comment by nan-li** [comment]
@yuyangpl You are able to resolve the badge issue now?

**Comment by yuyangpl** [comment]
yes发自我的 iPhone在 2025年6月11日，02:26，Nan ***@***.***> 写道：﻿nan-li left a comment (OneSignal/OneSignal-Flutter-SDK#893)
@yuyangpl You are able to resolve the badge issue now?
